GeneratorInterface/GenFilters/src/PythiaDauFilter.cc

```diff
@@ -99,9 +103,7 @@ bool PythiaDauFilter::filter(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	   ++ndau;
 	   for( unsigned int i=0; i<dauIDs.size(); ++i) {
 	     int IDanti = -dauIDs[i];
-	     int pythiaCode = PYCOMP(dauIDs[i]);
-	     int has_antipart = pydat2.kchg[3-1][pythiaCode-1];
-	     if( has_antipart == 0 ) IDanti = dauIDs[i];
+             if ( !(fLookupGen->particleData.isParticle( IDanti )) ) IDanti = dauIDs[i];
```

Comment:
Do we know that it is safe to call `fLookupGen->particleData.isParticle()` simultaneously from different threads?

Reply:
I just checked, `isParticle()` is not const:
https://github.com/mortenpi/pythia8/blob/47798bf862125fb03a70590198258a80cc1f11fd/include/ParticleData.h#L412

In theory, it could update the map internal to Pythia8. However, the code does first check if it can find the particle and only if it can does it go to the next line which only updates the map if the key wasn't there.

Reply:
If I understand correctly to modify the map, one has to use https://github.com/mortenpi/pythia8/blob/47798bf862125fb03a70590198258a80cc1f11fd/include/ParticleData.h#L106
in which case, hasAnti (L193) is const.